framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java

```diff
@@ -262,7 +315,8 @@ public void run() {
   }
   
   public synchronized void setTestStatus(TestStatus status) {
-	if (testStatus == TestStatus.CANCELED) return;
+	if (testStatus == TestStatus.CANCELED || testStatus == TestStatus.VERIFICATION_FAILURE) 
```

Comment:
You are right for the current setup. But it doesn't prevent one day others might change the framework and make the checking necessary. In any event, this is another defensive coding.